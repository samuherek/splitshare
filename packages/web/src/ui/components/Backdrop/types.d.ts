import React from 'react';
import { StandardProps } from '../types';
import { TransitionProps } from '../../core/utils/transitions';

export interface BackdropProps
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>> {
  /**
   * If `true`, the backdrop is invisible.
   * It can be used when rendering a popover or a custom select component.
   */
  invisible?: boolean;
  /**
   * If `true`, the backdrop is open.
   */
  open: boolean;
  /**
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   */
  transitionDuration?: TransitionProps['timeout'];
}
