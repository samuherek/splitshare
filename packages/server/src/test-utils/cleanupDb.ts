import { getConnection } from 'typeorm';

export default async function cleanupDb() {
  await getConnection().query('DELETE FROM bill_user');
  await getConnection().query('DELETE FROM receipt');
  await getConnection().query('DELETE FROM bill_invite');
  await getConnection().query('DELETE FROM bill');
  await getConnection().query('DELETE FROM "user"');
}
