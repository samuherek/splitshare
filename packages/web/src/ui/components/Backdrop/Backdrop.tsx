// @flow
import clsx from 'clsx';
import React from 'react';
import styled from 'styled-components';
import Fade from '../../transitions/Fade';
import { BackdropProps } from './types.d';

export const classes = {
  root: 'Backdrop',
};

const BackdropStyled = styled.div`
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: -1;
  position: fixed;
  touch-action: none;
`;

export const BackdropLight = styled.div<{ invisible?: boolean }>`
  background-color: ${({ invisible }) =>
    invisible ? 'transparent' : 'rgba(255,255,255,0.75)'};
`;

export const BackdropDark = styled.div<{ invisible?: boolean }>`
  background-color: ${({ invisible }) =>
    invisible ? 'transparent' : 'rgba(50,50,50,0.65)'};
`;

const Backdrop = React.forwardRef<any, BackdropProps>((props, ref) => {
  const {
    open,
    transitionDuration,
    invisible = false,
    className,
    ...rest
  } = props;

  return (
    <Fade in={open} timeout={transitionDuration} {...rest}>
      <BackdropStyled
        as={BackdropDark}
        className={clsx(className, classes.root)}
        aria-hidden
        // @ts-ignore
        invisible={invisible}
      />
    </Fade>
  );
});

export default Backdrop;
