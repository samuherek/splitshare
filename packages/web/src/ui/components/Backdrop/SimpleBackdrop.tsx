import React from 'react';
import styled from 'styled-components';
import { BackdropProps } from './types.d';

const WrapStyled = styled.div`
  z-index: -1;
  position: fixed;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.5);
  -webkit-tap-highlight-color-color: transparent;
`;

const SimpleBackdrop = React.forwardRef<unknown, BackdropProps>(
  function SimpleBackdrop(props, ref) {
    const { invisible = false, open, ...other } = props;

    return open ? (
      <WrapStyled
        aria-hidden={true}
        // FIXME: not sure how to type this
        // @ts-ignore
        ref={ref}
        style={invisible ? { backgroundColor: 'transparent' } : undefined}
        {...other}
      />
    ) : null;
  }
);

export default SimpleBackdrop;
