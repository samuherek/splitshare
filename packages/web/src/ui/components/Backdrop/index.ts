export { default } from './Backdrop';
export { default as SimpleBackdrop } from './SimpleBackdrop';
export * from './types.d';
