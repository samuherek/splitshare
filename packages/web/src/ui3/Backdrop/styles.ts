// @flow
import styled from 'styled-components';

export const BackdropLight = styled.div<{ invisible?: boolean }>`
  background-color: ${({ invisible }) =>
    invisible ? 'transparent' : 'rgba(255,255,255,0.75)'};
`;

export const BackdropDark = styled.div<{ invisible?: boolean }>`
  background-color: ${({ invisible }) =>
    invisible ? 'transparent' : 'rgba(50,50,50,0.65)'};
`;
