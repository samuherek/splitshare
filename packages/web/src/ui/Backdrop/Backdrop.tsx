// @flow
import clsx from 'clsx';
import React, { SyntheticEvent } from 'react';
import styled from 'styled-components';
import Fade from '../Fade';
import { BackdropDark, BackdropLight } from './styles';

interface Props {
  isOpen: boolean;
  duration: number;
  onClick?: (ev: SyntheticEvent<HTMLElement>) => void;
  invisible?: boolean;
  variant?: 'light' | 'dark';
  className?: string;
}

export const classes = {
  root: 'Backdrop',
};

const BackdropStyled = styled.div`
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: -1;
  position: fixed;
  touch-action: none;
`;

const variantComponent = {
  light: BackdropLight,
  dark: BackdropDark,
};

const Backdrop = React.forwardRef<Props, any>((props, ref) => {
  const {
    isOpen,
    duration,
    onClick,
    invisible = false,
    variant = 'dark',
    className,
    ...rest
  } = props;

  // @ts-ignore
  const Component = variantComponent[variant];

  return (
    <Fade in={isOpen} duration={duration} {...rest}>
      <BackdropStyled
        as={Component}
        className={clsx(className, classes.root)}
        aria-hidden
        // @ts-ignore
        invisible={invisible}
        onClick={onClick}
      />
    </Fade>
  );
});

export default Backdrop;
