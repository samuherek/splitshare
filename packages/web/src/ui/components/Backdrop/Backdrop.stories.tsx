import React from 'react';
import Backdrop from './Backdrop';

export default {
  title: 'Misc/Backdrop',
  component: Backdrop,
};

export const Base = () => {
  return (
    <>
      <div>Some content behind the backdrop</div>
      <Backdrop open={true} />
    </>
  );
};
