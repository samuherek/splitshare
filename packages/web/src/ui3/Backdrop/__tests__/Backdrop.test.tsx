import React from 'react';
import { render } from '../../../test-utils/test-utils';
import Backdrop from '../Backdrop';

describe('<Backdrop />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Backdrop />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
