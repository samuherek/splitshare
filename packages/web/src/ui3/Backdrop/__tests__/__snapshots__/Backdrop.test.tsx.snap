// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> should match snapshot 1`] = `
.c0 {
  background-color: rgba(50,50,50,0.65);
}

.c1 {
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: -1;
  position: fixed;
  touch-action: none;
}

<div
  aria-hidden="true"
  class="c0 c1 Backdrop"
/>
`;
